### PR TITLE
fix: instance network interface mutex sync

### PR DESCRIPTION
**Pull request by deepaksibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by hkantare** [comment]
For any PR please run respective testcase

**Comment by hkantare** [comment]
For any PR please run respective testcase

**Comment by deepaksibm** [comment]
> For any PR please run respective testcase
```
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsInstanceNetworkInterface_rip'
=== RUN   TestAccIBMIsInstanceNetworkInterface_rip
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
--- PASS: TestAccIBMIsInstanceNetworkInterface_rip (268.53s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     269.490s
```
```
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsInstanceNetworkInterfaceAllArgs'
=== RUN   TestAccIBMIsInstanceNetworkInterfaceAllArgs
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
--- PASS: TestAccIBMIsInstanceNetworkInterfaceAllArgs (328.22s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     329.596s
```

File: ibm/service/vpc/resource_ibm_is_instance_network_interface.go [hunk 1]
```diff
@@ -621,7 +621,7 @@ func resourceIBMIsInstanceNetworkInterfaceUpdate(context context.Context, d *sch
 		hasChange = true
 	}
 	if hasChange {
-		isNICKey := "instance_network_interface_key_" + instance_id
+		isNICKey := "instance_key_" + instance_id
```
[the remark this answers is not in the document]
Comment (deepaksibm) [reply ibm/service/vpc/resource_ibm_is_instance_network_interface.go:624]: yes, when the key was different we were getting conflicts while deleting, as the mutex was on different keys. so modified the key to be in sync with instance and instance related resources

[the remark this answers is not in the document]
Comment (deepaksibm) [reply ibm/service/vpc/resource_ibm_is_instance_network_interface.go:624]: @hkantare is this comment good to resolve ?

**Pull request merged**